mlflow/utils/string_utils.py

```diff
@@ -29,3 +29,20 @@ def truncate_str_from_middle(s, max_length):
         left_part_len = (max_length - 3) // 2
         right_part_len = max_length - 3 - left_part_len
         return f"{s[:left_part_len]}...{s[-right_part_len:]}"
+
+
+def dedup_string_list(str_list):
+    """
+    De-duplicate a list of strings. For duplicated strings, add suffix such as (1), (2), etc.
```

Comment:
This docstring indicates this function converts `["xb", "xb"]` to `["xb(1)", "xb(2)"]`, but it doesn't. Can you fix it?

Reply:
I'd prefer `["xb(1)", "xb(2)"]`

Reply:
Considering https://github.com/mlflow/mlflow/pull/6705#issuecomment-1237618055

I propose in this PR to change the suffix to be `({f_{feature_index})` instead of `({dup_index})`